nixos/modules/services/security/opensnitch.nix

```
@@ -0,0 +1,174 @@
+        default = {
+          default_timeout = 60;
+          default_action = "deny";
+          default_duration = "once";
+        };
+        example = ''
+          {
+            default_timeout = 15;
+            default_action = "allow";
+            default_duration = "until restart";
+          }
+        '';
+      };
+
+      startUserService = mkOption {
+        type = types.bool;
+        description = ''
+          If enabled, run the opensnitch-ui process as a user
+          service in the graphical session.  All users' UI processes
+          share the config specified in uiConfig.  If this is disabled,
+          opensnitch-ui must be run by some other means.
+        '';
+        default = true;
+      };
+
+      whitelistPackages = mkOption {
+        type = types.listOf types.package;
+        description = ''
+          List of packages for which to generate rules to allow connections
+          from all processes that are located below
+          a package's store path.  Intended for process rules
+          which should survive NixOS updates.
+        '';
+        default = with pkgs; [ nix ];
+      };
+
+      whitelistHosts = mkOption {
+        type = types.listOf types.attrs;
+        description = ''
+          List of destination hosts for which to create default (regexp) allow rules,
+          regardless of other connection properties.
+        '';
+        default = [];
+        example = ''[ { host = "*.nixos.org" } { ip = "127.0.0.1" } ]'';
+      };
+
+      extraRules = mkOption {
+        type = types.attrsOf types.attrs;
+        description = ''
+          Set of JSON attribute sets describing default opensnitch rules that are written to
+          /etc/opensnitch/rules.
+        '';
+        default = [];
+        example = ''
+          {
+            kerberos_rule = {
+              action = "allow";
+              operator = {
+                type = "list";
+                list = [
+                  {
+                    type = "simple";
+                    operand = "dest.host";
+                    data = "kerberos.example.domain";
+                  }
+                  {
+                    type = "simple";
+                    operand = "dest.port";
+                    data = "88";
+                  }
+                ];
+              };
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config =
+    let
+      uiConfig = writeText "ui-config.json" (builtins.toJSON cfg.uiConfig);
+    in
+    lib.mkIf cfg.enable {
+
+      environment.etc = listToAttrs (
+        (map (makePackageRuleFile "opensnitchd/rules") cfg.whitelistPackages)
+        ++ concatMap (x: mapAttrsToList (n: v: makeHostRuleFile "opensnitchd/rules" n v) x) cfg.whitelistHosts
+        ++ mapAttrsToList (makeAlwaysRuleFile "opensnitchd/rules") cfg.extraRules
+      );
+
+      environment.systemPackages = if cfg.startUserService then [] else [ opensnitch-ui ];
```

```suggestion
      environment.systemPackages = mkIf (!cfg.startUserService) [ opensnitch-ui ];
```

Or `optional`/`optionals`.